src/bidiMapper/mapper.ts

```diff
@@ -68,18 +68,24 @@ const _waitSelfTargetIdPromise = _waitSelfTargetId();
 function _createCdpClient() {
   // A CdpTransport implementation that uses the window.cdp bindings
   // injected by Target.exposeDevToolsProtocol.
-  class WindowCdpTransport implements CdpTransport {
-    onmessage?: (message: string) => void;
+  class WindowCdpTransport implements IServer {
+    private _onMessage?: (message: string) => void;
 
     constructor() {
       window.cdp.onmessage = (message: string) => {
-        if (this.onmessage) {
-          this.onmessage.call(null, message);
+        if (this._onMessage) {
+          this._onMessage.call(null, message);
         }
       };
     }
 
-    sendMessage(message: string): void {
+    async setOnMessage(
+      onMessage: (messageObj: any) => Promise<void>
+    ): Promise<void> {
+      this._onMessage = onMessage;
+    }
+
+    async sendMessage(message: string): Promise<void> {
```

Comment:
You added `async` here, but the implementation doesn't `await`. Is TypeScript different from JS, or is that just missing?

Reply:
the implementation of `window.cdp.send` is synchronous, but the generic method `sendMessage` of `IServer` can be async, so instead of returning resolved promise, I marked it as async.